robotbase/src/main/java/com/team2357/frc2022/subsystems/ClimberSubsystem.java

```diff
@@ -41,17 +62,82 @@
-        m_leftClimberMotor.setIdleMode(m_config.m_climberMotorIdleMode);
-        m_leftClimberMotor.setSmartCurrentLimit(m_config.m_climberMotorStallLimitAmps,
-                m_config.m_climberMotorFreeLimitAmps);
+        configureClimberMotor(m_leftClimberMotor);
+        configureClimberMotor(m_rightClimberMotor);
+
+        m_leftPidController = m_leftClimberMotor.getPIDController();
+        configureClimberPID(m_leftPidController);
 
+        m_rightPidController = m_rightClimberMotor.getPIDController();
+        configureClimberPID(m_rightPidController);
+
+        m_leftClimberMotor.setInverted(!m_config.m_isRightSideInverted);
         m_rightClimberMotor.setInverted(m_config.m_isRightSideInverted);
-        m_rightClimberMotor.setIdleMode(m_config.m_climberMotorIdleMode);
-        m_rightClimberMotor.setSmartCurrentLimit(m_config.m_climberMotorStallLimitAmps,
+    }
+
+    private void configureClimberMotor(CANSparkMax motor) {
+        motor.setIdleMode(m_config.m_climberMotorIdleMode);
+        motor.setSmartCurrentLimit(m_config.m_climberMotorStallLimitAmps,
                 m_config.m_climberMotorFreeLimitAmps);
     }
 
+    private void configureClimberPID(SparkMaxPIDController pidController) {
+        // set PID coefficients
+        pidController.setP(m_config.m_climberMotorP);
+        pidController.setI(m_config.m_climberMotorI);
+        pidController.setD(m_config.m_climberMotorD);
+        pidController.setIZone(m_config.m_climberMotorIZone);
+        pidController.setFF(m_config.m_climberMotorFF);
+        pidController.setOutputRange(m_config.m_climberMotorMinOutput, m_config.m_climberMotorMinOutput);
+
+        // Configure smart motion
+        int smartMotionSlot = 0;
+        pidController.setSmartMotionMaxVelocity(m_config.m_climberMotorMaxVel, smartMotionSlot);
+        pidController.setSmartMotionMinOutputVelocity(m_config.m_climberMotorMinVel, smartMotionSlot);
+        pidController.setSmartMotionMaxAccel(m_config.m_climberMotorMaxAcc, smartMotionSlot);
+        pidController.setSmartMotionAllowedClosedLoopError(m_config.m_climberMotorAllowedError, smartMotionSlot);
+    }
+
+    /**
+     * 
+     * @param rotations Setpoint in rotations for the climber motor to go to
+     * @return
+     */
+    public void setClimberRotations(double rotations) {
+        m_targetRotations = rotations;
+        m_leftPidController.setReference(m_targetRotations, CANSparkMax.ControlType.kSmartMotion);
+        m_rightPidController.setReference(m_targetRotations, CANSparkMax.ControlType.kSmartMotion);
+    }
+
+    /**
+     * 
+     * @return Is the climber at the setpoint set by {@Link setClimberRotations}
+     */
+    public boolean isClimberAtRotations() {
+        
+        return isLeftClimberAtRotations() && isRightClimberAtRotations();
+    }
+
+    /**
+     * 
+     * @return Is the climber at the setpoint set by {@Link setClimberRotations}
+     */
+    public boolean isLeftClimberAtRotations() {
```

Comment:
Nitpick here: you could have this function take the climber motor object and then you only need 1 function instead of two.